### PR TITLE
[chore] Split core and contrib workflows

### PR DESCRIPTION
**Pull request by TylerHelmuth**
This PR separates the Core and Contrib distributions into separate workflows.

The repository continues to use a single goreleaser.yaml file because:
1. we already have a process to generate it
2. goreleaser has features to support a single file

We specifically take advantage of `id` to specify exactly which things in the goreleaser.yaml file get built.

For now I did not change the triggers - both sets of workflows will currently always be triggered at the same time, like they are today.

Related to https://github.com/open-telemetry/opentelemetry-collector-releases/issues/429

Example workflows: 
- https://github.com/TylerHelmuth/opentelemetry-collector-releases/actions/runs/7908747507
- https://github.com/TylerHelmuth/opentelemetry-collector-releases/actions/runs/7908747499

I couldn't test the actual release bc I don't have a goreleaser-pro key.

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
Created https://github.com/open-telemetry/opentelemetry-collector-releases/pull/478 to take care of ensuring `partial` is included in the generate.

File: .github/workflows/ci-goreleaser-core.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: Continuous Integration - GoReleaser - Core
```
Comment (jpkrohling) [inline .github/workflows/ci-goreleaser-core.yaml:1]: I did some similar work in this repo, with reusable workflows: https://github.com/jpkrohling/otelcol-distributions

Basically, I defined the generic part here: https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/base-release.yaml

And then each distribution would just pass parameters:
- https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/release-loadbalancing.yaml
- https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/release-sidecar.yaml
- https://github.com/jpkrohling/otelcol-distributions/blob/main/.github/workflows/release-tracing.yaml

Comment (TylerHelmuth) [reply .github/workflows/ci-goreleaser-core.yaml:1]: @jpkrohling perfect, I can implement this reusable pattern now instead of later

Comment (TylerHelmuth) [reply .github/workflows/ci-goreleaser-core.yaml]: Done

File: .github/workflows/base-release.yaml [hunk 2]
```diff
@@ -133,8 +135,8 @@ jobs:
         with:
           distribution: goreleaser-pro
           version: latest
-          args: continue --merge --timeout 2h
+          args: continue --merge --timeout 2h --id ${{ inputs.distribution }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          COSIGN_YES: true
```
Comment (jpkrohling) [inline .github/workflows/base-release.yaml:139]: Can you double-check this? I think my version is older than what we have in this repo here when it comes to cosign.

File: .github/workflows/base-release.yaml [hunk 3]
```diff
@@ -98,19 +100,19 @@ jobs:
...
       - uses: docker/setup-qemu-action@v3
         with:
           platforms: arm64,ppc64le,s390x
 
       - uses: docker/setup-buildx-action@v3
 
-      - uses: actions/setup-go@v5
+      - uses: actions/setup-go@v4
         with:
-          go-version: '~1.21.7'
```
Comment (jpkrohling) [inline .github/workflows/base-release.yaml:113]: Can you double-check this change? Is this downgrade intentional?

File: .github/workflows/base-release.yaml [hunk 4]
```diff
@@ -98,19 +100,19 @@ jobs:
         with:
           fetch-depth: 0
 
-      - uses: sigstore/cosign-installer@v3
```
Comment (jpkrohling) [inline .github/workflows/base-release.yaml:101]: Can you check these downgrades? Were they intentional? Perhaps my version is older than that we have here in this repo

File: .github/workflows/base-release.yaml [hunk 5]
```diff
@@ -70,12 +72,12 @@ jobs:
         with:
           distribution: goreleaser-pro
           version: latest
-          args: release --clean --split --timeout 2h
+          args: release --clean --split --timeout 2h --id $ {{ inputs.distribution }}
         env:
           GOOS: ${{ matrix.GOOS }}
           GOARCH: ${{ matrix.GOARCH }}
           GITHUB_TOKEN: ${{ secrets.GH_PAT }}
-          COSIGN_YES: true
```
Comment (jpkrohling) [inline .github/workflows/base-release.yaml:78]: This also needs double-checking

**Review by jpkrohling — COMMENTED** [review]
It looks mostly right, but there are a few cosign-related changes that seem misplaced.

**Comment by TylerHelmuth** [comment]
@jpkrohling all those downgrades were the result of a bad merge.  Should now be fixed.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**